c2rust-analyze/src/main.rs

```diff
@@ -576,6 +564,35 @@ fn run(tcx: TyCtxt) {
     }
     eprintln!("reached fixpoint in {} iterations", loop_count);
 
+    // For testing, putting #[c2rust_analyze_test::fail_before_rewriting] on a function marks it as
+    // failed at this point.
+    for &ldid in &all_fn_ldids {
+        if !util::has_test_attr(tcx, ldid, TestAttr::FailBeforeRewriting) {
+            continue;
+        }
+        gacx.mark_fn_failed(
+            ldid.to_def_id(),
+            PanicDetail::new("explicit fail_before_rewriting for testing".to_owned()),
+        );
+    }
+
+    // Before generating rewrites, add the FIXED flag to the signatures of all functions that
+    // failed analysis.
+    for did in gacx.iter_fns_failed() {
+        let lsig = match gacx.fn_sigs.get(&did) {
+            Some(x) => x,
+            None => continue,
+        };
+        for sig_lty in lsig.inputs.iter().copied().chain(iter::once(lsig.output)) {
```

Comment:
nit: A `.inputs_and_output()` method would be nice.  I think the `rustc` `FnSig` equivalent has something similar.